skbio/sequence/_sequence.py

```diff
@@ -987,6 +945,231 @@ def _format_str(self, s):
+        >>> pprint(seq.metadata)
+        {'authors': ['Alice'], 'id': 'seq-id'}
+
+        The same applies to the sequence's positional metadata:
+
+        >>> seq_copy.positional_metadata.loc[0, 'quality'] = 999
+        >>> seq_copy.positional_metadata
+          list  quality
+        0   []      999
+        1   []       10
+        2   []        8
+        3   []        5
+        >>> seq.positional_metadata
+          list  quality
+        0   []        7
+        1   []       10
+        2   []        8
+        3   []        5
+
+        Since only a *shallow* copy was made, updates to mutable objects stored
+        as metadata affect the original sequence's metadata:
+
+        >>> seq_copy.metadata['authors'].append('Bob')
+        >>> pprint(seq_copy.metadata)
+        {'authors': ['Alice', 'Bob'], 'id': 'new-id'}
+        >>> pprint(seq.metadata)
+        {'authors': ['Alice', 'Bob'], 'id': 'seq-id'}
+
+        The same applies to the sequence's positional metadata:
+
+        >>> seq_copy.positional_metadata.loc[0, 'list'].append(1)
+        >>> seq_copy.positional_metadata
+          list  quality
+        0  [1]      999
+        1   []       10
+        2   []        8
+        3   []        5
+        >>> seq.positional_metadata
+          list  quality
+        0  [1]        7
+        1   []       10
+        2   []        8
+        3   []        5
+
+        Perform a deep copy to avoid this behavior:
+
+        >>> seq_deep_copy = seq.copy(deep=True)
+
+        Updates to mutable objects no longer affect the original sequence's
+        metadata:
+
+        >>> seq_deep_copy.metadata['authors'].append('Carol')
+        >>> pprint(seq_deep_copy.metadata)
+        {'authors': ['Alice', 'Bob', 'Carol'], 'id': 'seq-id'}
+        >>> pprint(seq.metadata)
+        {'authors': ['Alice', 'Bob'], 'id': 'seq-id'}
+
+        Nor its positional metadata:
+
+        >>> seq_deep_copy.positional_metadata.loc[0, 'list'].append(2)
+        >>> seq_deep_copy.positional_metadata
+             list  quality
+        0  [1, 2]        7
+        1      []       10
+        2      []        8
+        3      []        5
+        >>> seq.positional_metadata
+          list  quality
+        0  [1]        7
+        1   []       10
+        2   []        8
+        3   []        5
+
+        """
+        return self._copy(deep, {})
+
+    def _copy(self, deep, memo):
+        # strategy: copy the sequence without metadata first, then set metadata
+        # attributes with copies. we take this approach instead of simply
+        # passing the metadata through the Sequence constructor because we
+        # don't want to copy twice (this could happen when deep=True, where we
+        # deep copy here and then shallow copy in the Sequence constructor). we
+        # also directly set the private metadata attributes instead of using
+        # their public setters to avoid an unnecessary copy
+        if deep:
```

Comment:
There shouldn't be any difference. I thought it'd be safer to use `deepcopy` in case the type changes in the future, but the performance hit probably isn't worth the defensive programming. I'll remove this branch and add a comment for why we're not using `deepcopy` here.